### PR TITLE
fix(auth): failing requests result in status_code 401

### LINKED ISSUE #924: bug(auth): requests to the api with an invalid token receive a response status 500 instead of 401

### PR DESCRIPTION
**Pull request by fugal-dy**
closes #924

### CODE REVIEW

File: timed/authentication.py [hunk 1]
```diff
@@ -33,22 +34,34 @@ def get_introspection(self, access_token, id_token, payload):
...
-            if "client_id" not in claims:
-                raise SuspiciousOperation("client_id not present in introspection")
+        while True:
+            try:
+                claims = self.cached_request(
+                    self.get_userinfo, access_token, "auth.userinfo"
+                )
+                break
+            except requests.HTTPError as e:
+                if not (e.response.status_code in [401, 403]):
```
Comment (winged) [inline timed/authentication.py:44]:
```suggestion
                if e.response.status_code not in [401, 403]:
```

File: timed/authentication.py [hunk 2]
```diff
@@ -33,22 +34,34 @@ def get_introspection(self, access_token, id_token, payload):
...
-            ):
-                raise e
-
-            # check introspection if userinfo fails (confidental client)
-            claims = self.cached_request(
-                self.get_introspection, access_token, "auth.introspection"
-            )
-            if "client_id" not in claims:
-                raise SuspiciousOperation("client_id not present in introspection")
+        while True:
```
Comment (winged) [inline timed/authentication.py:37]: Why the loop? There is no condition where the code ever *actually* loops

File: timed/authentication.py [hunk 3]
```diff
@@ -33,22 +34,34 @@ def get_introspection(self, access_token, id_token, payload):
...
+            try:
+                claims = self.cached_request(
+                    self.get_userinfo, access_token, "auth.userinfo"
+                )
+                break
```
Comment (winged) [inline timed/authentication.py:42]: you could just early-return here
```suggestion
                return claims
```

File: timed/authentication.py [hunk 4]
```diff
@@ -33,22 +34,34 @@ def get_introspection(self, access_token, id_token, payload):
...
+                    try:
+                        # check introspection if userinfo fails (confidential client)
+                        claims = self.cached_request(
+                            self.get_introspection, access_token, "auth.introspection"
+                        )
+                        if "client_id" not in claims:
+                            raise SuspiciousOperation(
+                                "client_id not present in introspection"
+                            )
+                        break
```
Comment (winged) [inline timed/authentication.py:56]:
```suggestion
                        return claims
```

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -9,7 +9,7 @@ repos:
       - id: isort
         name: isort
         language: system
-        entry: isort -y
+        entry: isort
```
Comment (winged) [inline .pre-commit-config.yaml:12]: Unrelated change - can you split it into a separate commit?

**Review by winged — CHANGES_REQUESTED** [review]

File: timed/authentication.py (hunk 2 above)
Comment (fugal-dy) [reply timed/authentication.py]: Actually to avoid returning early ;)

and only once.

Changed it.

File: .pre-commit-config.yaml (hunk 5 above)
Comment (fugal-dy) [reply .pre-commit-config.yaml:12]: i did.

**Review by winged — APPROVED** [review]
